WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt

```diff
@@ -0,0 +1,48 @@
+package org.wordpress.android.modules
+
+import android.content.Context
+
+import com.android.volley.RequestQueue
+import com.bumptech.glide.Glide
+import com.bumptech.glide.GlideBuilder
+import com.bumptech.glide.Registry
+import com.bumptech.glide.annotation.GlideModule
+import com.bumptech.glide.integration.volley.VolleyUrlLoader
+import com.bumptech.glide.load.model.GlideUrl
+import com.bumptech.glide.module.AppGlideModule
+
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.network.HTTPAuthManager
+import org.wordpress.android.fluxc.network.UserAgent
+import org.wordpress.android.fluxc.network.rest.wpcom.auth.AccessToken
+import org.wordpress.android.networking.GlideHeaderLoaderFactory
+
+import java.io.InputStream
+
+import javax.inject.Inject
+import javax.inject.Named
+
+/**
+ * Custom [AppGlideModule] that replaces Glide's default [RequestQueue] with FluxC's and sets custom GlideHeaderLoader
+ * which adds support for custom authorization headers.
+ */
+@GlideModule
+class WordPressGlideModule : AppGlideModule() {
+    @Inject @field:Named("custom-ssl") lateinit var mRequestQueue: RequestQueue
+    @Inject lateinit var mAccessToken: AccessToken
+    @Inject lateinit var mHttpAuthManager: HTTPAuthManager
+    @Inject lateinit var mUserAgent: UserAgent
+
+    override fun applyOptions(context: Context, builder: GlideBuilder) {}
+
+    override fun isManifestParsingEnabled(): Boolean {
+        return false
+    }
+
+    override fun registerComponents(context: Context, glide: Glide, registry: Registry) {
+        (context as WordPress).component().inject(this)
+        registry.replace(GlideUrl::class.java, InputStream::class.java, VolleyUrlLoader.Factory(mRequestQueue))
+        registry.append(GlideUrl::class.java, InputStream::class.java,
```

Comment:
This means that WordPress.com urls running through Glide will first attempt an un-authed connection, then try authed.

So, for example, loading a private image URL on a WordPress.com site through Glide will generate at least 3 requests:

1. A failed, un-authed request to download the image
2. A single retry of the above connection, also fails
3. An authed attempt to download the image

Probably there was a reason not to `replace`, but there should be a way to let the `GliderHeaderLoader` be the only header manager for Glide.